docker_images/sklearn/tests/generators/generate.py

```diff
@@ -0,0 +1,229 @@
+    # different sklearn versions.
+    yield "logistic_regression", make_pipeline(StandardScaler(), LogisticRegression())
+
+    # this estimator cannot be loaded on 1.1 if it's stored using 1.0, but it
+    # handles NaN input values which the previous pipeline cannot handle.
+    yield "hist_gradient_boosting", HistGradientBoostingClassifier()
+
+
+def get_tabular_regressors():
+    # yield regressor names and estimators to train and push to hub.
+
+    # this is a pipeline with simple estimators which can be loaded across
+    # different sklearn versions.
+    yield "linear_regression", make_pipeline(StandardScaler(), LinearRegression())
+
+    # this estimator cannot be loaded on 1.1 if it's stored using 1.0, but it
+    # handles NaN input values which the previous pipeline cannot handle.
+    yield "hist_gradient_boosting_regressor", HistGradientBoostingRegressor()
+
+
+def create_repos(est_name, task_name, est, sample, version):
+    # given trained estimator instance, it's name, and the version tag, push to
+    # hub once with and once without a config file.
+
+    # initialize repo
+    _, pkl_name = mkstemp(prefix="skops-", suffix=".pkl")
+    with open(pkl_name, mode="bw") as f:
+        pickle.dump(est, file=f)
+
+    local_repo = mkdtemp(prefix="skops-")
+    hub_utils.init(
+        model=pkl_name,
+        requirements=[f"scikit-learn={sklearn.__version__}"],
+        dst=local_repo,
+        task=task_name,
+        data=sample,
+    )
+
+    # push WITH config
+    repo_name = REPO_NAMES[task_name].format(
+        version=version, est_name=est_name, w_or_wo="with"
+    )
+    push_repo(repo_name=repo_name, local_repo=local_repo)
+
+    # push WIHTOUT CONFIG
+    repo_name = REPO_NAMES[task_name].format(
+        version=version, est_name=est_name, w_or_wo="without"
+    )
+
+    # Now we remove the config file and push to a new repo
+    os.remove(Path(local_repo) / "config.json")
+    # The only valid file name for a model pickle file if no config.json is
+    # available is `sklearn_model.joblib`, otherwise the backend will fail to
+    # find the file.
+    os.rename(Path(local_repo) / pkl_name, Path(local_repo) / "sklearn_model.joblib")
+
+    push_repo(
+        repo_name=repo_name,
+        local_repo=local_repo,
+    )
+
+
+def save_sample(sample, filename):
+    payload = {"data": sample.to_dict(orient="list")}
+    with open(Path(__file__).parent / "samples" / filename, "w") as f:
+        json.dump(payload, f, indent=2)
+
+
+def predict_tabular_classifier(est, sample, filename):
+    output = [int(x) for x in est.predict(sample)]
+    with open(Path(__file__).parent / "samples" / filename, "w") as f:
+        json.dump(output, f, indent=2)
+
+
+def predict_tabular_regressor(est, sample, filename):
+    output = [float(x) for x in est.predict(sample)]
+    with open(Path(__file__).parent / "samples" / filename, "w") as f:
+        json.dump(output, f, indent=2)
+
+
+#############
+# CONSTANTS #
+#############
```

Comment:
move to top of the file?

Reply:
Doesn't work since the constants partly reference functions that need to be defined before.